tests/unit/test_in_mem_pub.py

```diff
@@ -16,44 +16,22 @@
 
 """Testing the in-memory publisher."""
 
-from contextlib import nullcontext
-
 import pytest
 
 from hexkit.providers.testing.in_mem_pub import (
     InMemEventPublisher,
     InMemEventStore,
     TopicExhaustedError,
 )
-from hexkit.utils import NonAsciiStrError
-
-
-@pytest.mark.parametrize(
-    "type_, key, topic, exception",
-    [
-        ("test_type", "test_key", "test_topic", None),
-        (
-            "test_ßtype",  # non ascii
-            "test_key",
-            "test_topic",
-            NonAsciiStrError,
-        ),
-        (
-            "test_type",
-            "test_ßkey",  # non ascii
-            "test_topic",
-            NonAsciiStrError,
-        ),
-        (
-            "test_type",
-            "test_key",
-            "test_ßtopic",  # non ascii
-            NonAsciiStrError,
-        ),
-    ],
-)
-def test_in_mem_publisher(type_, key, topic, exception):
+
+
+@pytest.mark.asyncio
+async def test_in_mem_publisher():
```

Comment:
See same comment for KafkaPublisher. The error tests may still make sense to check that super() is used by the InMemPublisher.